include/lsst/afw/table/python/catalog.h

```diff
@@ -167,7 +167,12 @@ PyCatalog<Record> declareCatalog(pybind11::module & mod, std::string const & nam
     });
     cls.def("_delslice_", [](Catalog & self, py::slice const & s) {
         Py_ssize_t start=0, stop=0, step=0, length=0;
-        if (PySlice_GetIndicesEx((PySliceObject*)s.ptr(), self.size(), &start, &stop, &step, &length) != 0) {
+        if (PySlice_GetIndicesEx(
+#if PY_MAJOR_VERSION < 3 || (PY_MINOR_VERSION == 3) && (PY_MINOR_VERSION < 2)
+            (PySliceObject*)
+#endif
+                s.ptr(), self.size(), &start, &stop, &step, &length) != 0
+        ) {
```

Comment:
I wish there was a slightly more readable way to do this, but it looks good enough to me. You might consider a brief comment saying "the API changed in Python 3.2"